internal/testgen/gen_bool_compare.go

```diff
@@ -26,6 +26,140 @@ func (g BoolCompareTestsGenerator) TemplateData() any {
+	for _, fn := range []string{
+		"Equal",
+		"EqualValues",
+		"Exactly",
+	} {
+		invalidAssertionsForTrue = append(invalidAssertionsForTrue,
+			Assertion{Fn: fn, Argsf: "predicate, true", ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: "predicate"},
+			Assertion{Fn: fn, Argsf: "true, predicate", ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: "predicate"},
+		)
+
+		invalidAssertionsForFalse = append(invalidAssertionsForFalse,
+			Assertion{Fn: fn, Argsf: "predicate, false", ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: "predicate"}, //nolint:lll
+			Assertion{Fn: fn, Argsf: "false, predicate", ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: "predicate"}, //nolint:lll
+		)
+	}
+
+	for _, fn := range []string{
+		"NotEqual",
+		"NotEqualValues",
+	} {
+		invalidAssertionsForTrue = append(invalidAssertionsForTrue,
+			Assertion{Fn: fn, Argsf: "predicate, false", ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: "predicate"},
+			Assertion{Fn: fn, Argsf: "false, predicate", ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: "predicate"},
+		)
+
+		invalidAssertionsForFalse = append(invalidAssertionsForFalse,
+			Assertion{Fn: fn, Argsf: "predicate, true", ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: "predicate"}, //nolint:lll
+			Assertion{Fn: fn, Argsf: "true, predicate", ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: "predicate"}, //nolint:lll
+		)
+	}
+
+	for _, argsf := range []string{
+		"predicate == false",
+		"false == predicate",
+		"predicate != true",
+		"true != predicate",
+		"!predicate",
+	} {
+		invalidAssertionsForTrue = append(invalidAssertionsForTrue,
+			Assertion{Fn: "False", Argsf: argsf, ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: "predicate"},
+		)
+
+		invalidAssertionsForFalse = append(invalidAssertionsForFalse,
+			Assertion{Fn: "True", Argsf: argsf, ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: "predicate"},
+		)
+	}
+
+	invalidAssertionsForTrue = append(invalidAssertionsForTrue,
+		Assertion{Fn: "False", Argsf: `!result["flag"].(bool)`, ReportMsgf: reportUse, ProposedFn: "True", ProposedArgsf: `result["flag"].(bool)`}, //nolint:lll
+	)
+
+	invalidAssertionsForFalse = append(invalidAssertionsForFalse,
+		Assertion{Fn: "True", Argsf: `!result["flag"].(bool)`, ReportMsgf: reportUse, ProposedFn: "False", ProposedArgsf: `result["flag"].(bool)`}, //nolint:lll
+	)
+
+	for _, argsf := range []string{
+		"predicate == true",
+		"true == predicate",
+		"predicate != false",
+		"false != predicate",
+	} {
+		invalidAssertionsForTrue = append(invalidAssertionsForTrue,
+			Assertion{Fn: "True", Argsf: argsf, ReportMsgf: reportSimplify, ProposedArgsf: "predicate"},
+		)
+
+		invalidAssertionsForFalse = append(invalidAssertionsForFalse,
+			Assertion{Fn: "False", Argsf: argsf, ReportMsgf: reportSimplify, ProposedArgsf: "predicate"},
+		)
+	}
+
+	sortAssertions(invalidAssertionsForTrue)
```

Comment:
any reason to sort it?
the order of assertions is determined

safe this PR from unrelated changes around generated test files

Reply:
I will remove it then